lib/ui/painting/image_decoder_impeller.cc

```diff
@@ -528,7 +575,14 @@ bool ImpellerAllocator::allocPixelRef(SkBitmap* bitmap) {
   descriptor.size = ((bitmap->height() - 1) * bitmap->rowBytes()) +
                     (bitmap->width() * bitmap->bytesPerPixel());
 
+#ifdef FML_OS_ANDROID
```

Comment:
Instead of having 2 conditional compilations you an use conditional compilation to set 1 const so that these 2 branches can be linked semantically

```c++
#ifdef FML_OS_ANDROID
static constexpr bool kShouldUseMallocDeviceBuffer = true;
#else
static constexpr bool kShouldUseMallocDeviceBuffer = false;
#endif
```